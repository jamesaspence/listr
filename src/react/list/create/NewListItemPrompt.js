import React, { useState } from 'react';
import './NewListItemPrompt.scss';

//TODO add editable state here

const NewListItemPrompt = ({ onNewItem }) => {
  const [ value, setValue ] = useState('');

  const onSubmit = event => {
    event.preventDefault();
    if (value.length < 1) {
      return;
    }
    onNewItem(value);
    setValue('');
  };

  const onTextChange = event => {
    setValue(event.target.value);
  };

  const isValidItemName = value.trim() !== '';
  const submitDisabledClassName = `NewListItemPrompt__submit ${isValidItemName ? '' : 'disabled'}`;

  return (
    <form className="NewListItemPrompt" onSubmit={onSubmit}>
      <input className="NewListItemPrompt__input" type="text" value={value} onChange={onTextChange} placeholder="...New item"/>
      <button className={submitDisabledClassName} type="submit" disabled={!isValidItemName}>Submit</button>
    </form>
  );
};

export default NewListItemPrompt;