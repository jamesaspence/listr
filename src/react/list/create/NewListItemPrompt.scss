@import '../../../variables';
@import '../../../mixins';

.NewListItemPrompt {
  grid-column: 1 / 5;
  padding: 12px;
  margin-bottom: 12px;
  border-bottom: 1px solid #afafaf;
  display: flex;
  flex-direction: column;
  text-align: center;

  &:hover {
    border-color: #7f7f7f;
  }

  .NewListItemPrompt__input, .NewListItemPrompt__submit {
    font-size: 16px;
    outline: none;
    border: none;
  }

  .NewListItemPrompt__input {
    background-color: transparent;
    flex: 1;
    padding: 6px 12px;
    text-align: center;
  }

  .NewListItemPrompt__submit {
    font-size: 16px;
    margin: 0 12px;
    padding: 12px;
    border-radius: 6px;
    cursor: pointer;

    &:not(.disabled) {
      background-color: #09cb9f;
      color: #fff;
    }

    &.disabled {
      pointer-events: none;
      background-color: #705959;
      color: #fff;
    }

    &:hover:not(.disabled) {
      opacity: 0.9;
    }
  }

  @include minWidth($size-mobile) {
    flex-direction: row;
    text-align: start;

    .NewListItemPrompt__input {
      text-align: initial;
    }
  }
}