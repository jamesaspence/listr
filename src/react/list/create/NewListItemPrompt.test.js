import { shallow } from 'enzyme';
import React from 'react';
import NewListItemPrompt from './NewListItemPrompt';

const onNewItem = jest.fn();

it('matches snapshot', () => {
  expect(shallow(<NewListItemPrompt onNewItem={onNewItem} />)).toMatchSnapshot();
});

it('does not trigger callback if value is empty', () => {
  const container = shallow(<NewListItemPrompt onNewItem={onNewItem}/>);

  container.find('.NewListItemPrompt__input').simulate('change', { target: { value: '' } });
  container.find('form').simulate('submit', new window.Event('submit'));

  expect(onNewItem.mock.calls.length).toEqual(0);
});

it('sends value to callback on submit', () => {
  const container = shallow(<NewListItemPrompt onNewItem={onNewItem}/>);
  const expectedText = 'oranges';

  container.find('.NewListItemPrompt__input').simulate('change', { target: { value: expectedText } });
  container.find('form').simulate('submit', new window.Event('submit'));

  expect(onNewItem.mock.calls.length).toEqual(1);
  expect(onNewItem.mock.calls[0][0]).toEqual(expectedText);
});

it('is disabled on empty string input', () => {
  const container = shallow(<NewListItemPrompt onNewItem={onNewItem}/>);

  container.find('.NewListItemPrompt__input').simulate('change', { target: { value: '   ' } });
  container.find('.NewListItemPrompt__submit').hasClass('disabled');
});