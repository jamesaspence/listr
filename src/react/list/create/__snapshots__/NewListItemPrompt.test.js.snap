// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  className="NewListItemPrompt"
  onSubmit={[Function]}
>
  <input
    className="NewListItemPrompt__input"
    onChange={[Function]}
    placeholder="...New item"
    type="text"
    value=""
  />
  <button
    className="NewListItemPrompt__submit disabled"
    disabled={true}
    type="submit"
  >
    Submit
  </button>
</form>
`;
